enterprise_access/apps/content_assignments/models.py

```diff
@@ -304,6 +304,17 @@ class Meta:
         blank=False,
         help_text="Cost of the content in USD Cents.",
     )
+    preferred_course_run_key = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        db_index=True,
+        help_text=(
+            "The preferred course run that the admin primarily intends for the learner, and the one used to control "
+            "nudge emails. This is automatically set at assignment creation time, and is never changed again (even "
+            "during re-allocation)."
```

Comment:
_Should_ we change it during re-allocation?

Reply:
The downside is that, if we don't change it during re-allocation, no nudges will go out for the assignment, right? I don't have strong feelings either way - maybe just add a note to the mgmt command's docstring.

Reply:
yea, this can happen:

* day 0: run A is the latest run, and admin assigns course.
* day 1: admin cancels assignment.
* day 2: run B is published.
* day 3: admin re-assigns the same learner to the same course.

Result: nudge emails will correspond to run A despite run B being the newest run at the time of re-assignment.

Reply:
after discussing with beggs, I'll just go ahead and add this feature in.